runtime/service.go

```diff
@@ -161,10 +163,16 @@ func NewService(shimCtx context.Context, id string, remotePublisher shim.Publish
 		namespace = namespaces.Default
 	}
 
+	var shimDir vm.Dir
 	vmID := os.Getenv(internal.VMIDEnvVarKey)
 	logger := log.G(shimCtx)
 	if vmID != "" {
 		logger = logger.WithField("vmID", vmID)
+
+		shimDir, err = vm.ShimDir(namespace, vmID)
+		if err != nil {
+			return nil, err
```

Comment:
nit: `return nil, errors.Wrap(err, "invalid shim directory")` or something similar for more context